api/modelmanager/modelmanager_test.go

```diff
@@ -325,45 +327,83 @@ func (s *modelmanagerSuite) TestUnsetModelDefaults(c *gc.C) {
+					{Tag: coretesting.ModelTag.String()},
+				},
+			})
+			c.Check(result, gc.FitsTypeOf, &params.ModelStatusResults{})
+
+			out := result.(*params.ModelStatusResults)
+			out.Results = []params.ModelStatus{
+				params.ModelStatus{
+					ModelTag:           coretesting.ModelTag.String(),
+					OwnerTag:           "user-glenda",
+					ApplicationCount:   3,
+					HostedMachineCount: 2,
+					Life:               "alive",
+					Machines: []params.ModelMachineInfo{{
+						Id:         "0",
+						InstanceId: "inst-ance",
+						Status:     "pending",
+					}},
+				},
+				params.ModelStatus{Error: common.ServerError(errors.New("model error"))},
+			}
+			return nil
+		},
+	}
+
+	client := modelmanager.NewClient(apiCaller)
+	results, err := client.ModelStatus(coretesting.ModelTag, coretesting.ModelTag)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(results[0], jc.DeepEquals, base.ModelStatus{
+		UUID:               coretesting.ModelTag.Id(),
+		TotalMachineCount:  1,
+		HostedMachineCount: 2,
+		ServiceCount:       3,
+		Owner:              "glenda",
+		Life:               string(params.Alive),
+		Machines:           []base.Machine{{Id: "0", InstanceId: "inst-ance", Status: "pending"}},
+	})
+	c.Assert(results[1].Error, gc.ErrorMatches, "model error")
+}
+
+func (s *modelmanagerSuite) TestModelStatusEmpty(c *gc.C) {
 	apiCaller := basetesting.APICallerFunc(func(objType string, version int, id, request string, arg, result interface{}) error {
 		c.Check(objType, gc.Equals, "ModelManager")
 		c.Check(id, gc.Equals, "")
 		c.Check(request, gc.Equals, "ModelStatus")
-		c.Check(arg, jc.DeepEquals, params.Entities{
-			[]params.Entity{{
-				Tag: coretesting.ModelTag.String(),
-			}},
-		})
 		c.Check(result, gc.FitsTypeOf, &params.ModelStatusResults{})
 
-		out := result.(*params.ModelStatusResults)
-		out.Results = []params.ModelStatus{{
-			ModelTag:           coretesting.ModelTag.String(),
-			OwnerTag:           "user-glenda",
-			ApplicationCount:   3,
-			HostedMachineCount: 2,
-			Life:               "alive",
-			Machines: []params.ModelMachineInfo{{
-				Id:         "0",
-				InstanceId: "inst-ance",
-				Status:     "pending",
-			}},
-		}}
 		return nil
 	})
 
 	client := modelmanager.NewClient(apiCaller)
-	results, err := client.ModelStatus(coretesting.ModelTag)
+	results, err := client.ModelStatus()
 	c.Assert(err, jc.ErrorIsNil)
-	c.Assert(results, jc.DeepEquals, []base.ModelStatus{{
-		UUID:               coretesting.ModelTag.Id(),
-		TotalMachineCount:  1,
-		HostedMachineCount: 2,
-		ServiceCount:       3,
-		Owner:              "glenda",
-		Life:               string(params.Alive),
-		Machines:           []base.Machine{{Id: "0", InstanceId: "inst-ance", Status: "pending"}},
-	}})
+	c.Assert(results, jc.DeepEquals, []base.ModelStatus{})
+}
+
+func (s *modelmanagerSuite) TestModelStatusError(c *gc.C) {
+	apiCaller := basetesting.BestVersionCaller{
+		BestVersion: 3,
+		APICallerFunc: basetesting.APICallerFunc(
```

Comment:
I'm surprised that this looks like a copy of the other TestModelStatusError, but has BestVersion: 3 while the other one doesn't have anything.